### PR TITLE
Allow configuring headers and URL for GitHub download

### PR DESCRIPTION
**Pull request by rmccue**
For private APIs (like GitHub's), this allows sending authentication tokens/etc.

e.g. to use a private repo:
```
	public function get_markdown_source( $post_id ) {
		$source = parent::get_markdown_source( $post_id );
		if ( is_wp_error( $source ) ) {
			return $source;
		}

		// Send requests to the GH REST API, with our token.
		$source = preg_replace( '#https?://github\.com/([^/]+/[^/]+)/blob/([^/]+)/(.+)#', 'https://api.github.com/repos/$1/contents/$3?ref=$2', $source );
		$source = add_query_arg( 'access_token', ACCESS_TOKEN, $source );
		return $source;
	}

	protected function get_markdown_source_headers( $id ) {
		return [
			'Accept' => 'application/vnd.github.v3.raw',
		];
	}
```

### CODE REVIEW

**Comment by rmccue** [comment]
Updated to also allow configuring the raw download URL separately to the source URL (which is used for user-facing links).

File: inc/class-importer.php [hunk 1]
```diff
@@ -330,4 +329,27 @@ public function get_markdown_source( $post_id ) {
...
+	}
+
+	/**
+	 * Get headers to send to Markdown source URL.
+	 *
+	 * @param int $post_id Post ID being fetched.
+	 * @return array Headers to pass to wp_remote_request()
+	 */
+	protected function get_markdown_source_headers( $post_id ) {
+		return array();
```
Comment (roborourke) [inline inc/class-importer.php:356]: Looks like this needs finishing off 👯‍♂️

Comment (roborourke) [reply inc/class-importer.php:356]: Lolz, ignore me!

Comment (rmccue) [reply inc/class-importer.php:356]: Yep, this is just here for subclass overrides. :)

**Pull request merged**